framework/util/image_writer.cpp

```diff
@@ -331,6 +334,73 @@ static const uint8_t* ConvertIntoTemporaryBuffer(uint32_t    width,
     return reinterpret_cast<const uint8_t*>(temporary_buffer.get());
 }
 
+static uint8_t* ExtractAlphaChannel(uint32_t width, uint32_t height, const void* data, uint32_t data_pitch, bool is_png)
```

Comment:
Could this be `expand_to_rgb` and invert the sense of the comparison?  I'd like if possible for parameters to generally express the lowest meaningful thing; if we were to add another luminance-capable image then a future reader would need to guess `is_png` means "can have one-channel encoding".

Reply:
sure